src/other/codemirror/CodeMirror.tsx

```diff
@@ -22,77 +25,93 @@ const LANGUAGES_MAP: Record<string, any> = {
 
 export interface CodeMirrorProps {
     updateContent: (content: string) => void;
-    updateOnFirstLoad: boolean;
     content?: string;
     options: boolean | BasicSetupOptions;
     language: string;
     completions: (context: CompletionContext) => CompletionResult;
     theme?: "light" | "dark";
-    onFocus?: () => void;
-    onBlur?: () => void;
+    checks?: ConsistencyCheck[];
     readOnly?: boolean;
 }
 
 export interface CodeMirrorState {
-    isLoaded: boolean;
+    content: string;
+    checks?: ConsistencyCheck[];
     isEditing: boolean;
 }
 
 class CodeMirror extends React.Component<CodeMirrorProps, CodeMirrorState> {
     constructor(props: CodeMirrorProps) {
         super(props);
-        this.state = { isLoaded: false, isEditing: false };
+        this.state = {
+            content: props.content || "",
+            checks: props.checks,
+            isEditing: false,
+        };
         this.handleContentChange = this.handleContentChange.bind(this);
     }
 
-    /*
-     * editor - CodeMirror object https://uiwjs.github.io/react-codemirror/
-     * viewUpdate - object containing the update to the editor tree structure
-     */
-    handleContentChange(newContent: string, viewUpdate: { origin: string }) {
-        const { isLoaded, isEditing } = this.state;
-        const { updateContent, updateOnFirstLoad = true } = this.props;
-        // kludge for the way state management is handled in web-app
-        if (!isLoaded && !updateOnFirstLoad && viewUpdate.origin === "setValue") {
-            this.setState({ isLoaded: true });
-            return;
+    UNSAFE_componentWillReceiveProps(nextProps: CodeMirrorProps) {
+        const { content, checks } = this.props;
+        if (nextProps.content !== content) {
+            this.setState({ content: nextProps.content || "" });
+        }
+
+        if (nextProps.checks !== checks) {
+            this.setState({ checks: nextProps.checks });
```

Comment:
Probably better to have the same condition as on line 73 and one `setState` accordingly

Reply:
Nope. This was a mistake to change. The logic that was here is correct and for a reason. Content and checks must be updated independetly